### PR TITLE
Selected routes

### PR DESCRIPTION
**Pull request by AbodHassam**

### CODE REVIEW

File: src/server/api/controllers/selected-routes.controller.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    return await knex("selected_routes").select("*");
+  } catch (err) {
+    err.message;
+  }
+};
+
+// Get a selected-route by voyage id
+const getSelectedRouteByVoyageId = async id => {
+  try {
+    const selected_routesByVoyageId = await knex("selected_routes")
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:18]: camelCase naming variables ;)

File: src/server/api/controllers/selected-routes.controller.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return selected_routesByVoyageId;
+  } catch (err) {
+    return err.message;
+  }
+};
+
+// Get the last selected-route for a given voyage
+const getlastSelectedRouteByVoyageId = async id => {
+  try {
+    const lastSelected_routesByVoyageId = await knex("selected_routes")
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:37]: Same here..

File: src/server/api/controllers/selected-routes.controller.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  }
+};
+
+// Get a selected-route by voyage id
+const getSelectedRouteByVoyageId = async id => {
+  try {
+    const selected_routesByVoyageId = await knex("selected_routes")
+      .select("*")
+      .where({ voyage_id: id });
+    if (selected_routesByVoyageId.length === 0) {
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:21]: same here

File: src/server/api/controllers/selected-routes.controller.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+// Get the last selected-route for a given voyage
+const getlastSelectedRouteByVoyageId = async id => {
+  try {
+    const lastSelected_routesByVoyageId = await knex("selected_routes")
+      .select("*")
+      .where({ voyage_id: id })
+      .orderBy("created_at", "desc")
+      .limit(1);
+    if (lastSelected_routesByVoyageId.length === 0) {
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:42]: same here

File: src/server/api/controllers/selected-routes.controller.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      throw new HttpError(
+        "Bad request",
+        `Cannot find selected routes for ID ${id}!`,
+        404
+      );
+    }
+    return lastSelected_routesByVoyageId;
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:49]: same here

File: src/server/api/controllers/selected-routes.controller.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      .select("*")
+      .where({ voyage_id: id });
+    if (selected_routesByVoyageId.length === 0) {
+      throw new HttpError(
+        "Bad request",
+        `Cannot find selected routes for ID ${id}!`,
+        404
+      );
+    }
+    return selected_routesByVoyageId;
```
Comment (ismailirahim) [inline src/server/api/controllers/selected-routes.controller.js:28]: same here

**Review by ismailirahim — CHANGES_REQUESTED** [review]
Change the variable naming to cameCase.
I also tested your endpoints and works fine  ;)

**Comment by zkwsk** [comment]
@AbodHassam I'm really sorry, but we realized that we need to change the DB design for selected_routes entirely, so I'm closing down this PR. I'll post some more info on this on the Slack channel shortly.

**Pull request closed without merging**